Compiled from "ProtocolConnectionUtils.java"
final class org.jboss.as.protocol.ProtocolConnectionUtils$WrapperCallbackHandler implements javax.security.auth.callback.CallbackHandler {
  private final org.jboss.as.protocol.GeneralTimeoutHandler timeoutHandler;

  private final javax.security.auth.callback.CallbackHandler wrapped;

  org.jboss.as.protocol.ProtocolConnectionUtils$WrapperCallbackHandler(org.jboss.as.protocol.GeneralTimeoutHandler, javax.security.auth.callback.CallbackHandler);
    Code:
       0: aload_0
       1: invokespecial #2                  // Method java/lang/Object."<init>":()V
       4: aload_0
       5: aload_1
       6: putfield      #3                  // Field timeoutHandler:Lorg/jboss/as/protocol/GeneralTimeoutHandler;
       9: aload_0
      10: aload_2
      11: putfield      #1                  // Field wrapped:Ljavax/security/auth/callback/CallbackHandler;
      14: return

  public void handle(javax.security.auth.callback.Callback[]) throws java.io.IOException, javax.security.auth.callback.UnsupportedCallbackException;
    Code:
       0: aload_0
       1: getfield      #3                  // Field timeoutHandler:Lorg/jboss/as/protocol/GeneralTimeoutHandler;
       4: new           #4                  // class org/jboss/as/protocol/ProtocolConnectionUtils$WrapperCallbackHandler$1
       7: dup
       8: aload_0
       9: aload_1
      10: invokespecial #5                  // Method org/jboss/as/protocol/ProtocolConnectionUtils$WrapperCallbackHandler$1."<init>":(Lorg/jboss/as/protocol/ProtocolConnectionUtils$WrapperCallbackHandler;[Ljavax/security/auth/callback/Callback;)V
      13: invokevirtual #6                  // Method org/jboss/as/protocol/GeneralTimeoutHandler.suspendAndExecute:(Ljava/lang/Runnable;)V
      16: goto          58
      19: astore_2
      20: aload_2
      21: invokevirtual #8                  // Method java/lang/RuntimeException.getCause:()Ljava/lang/Throwable;
      24: instanceof    #9                  // class java/io/IOException
      27: ifeq          38
      30: aload_2
      31: invokevirtual #8                  // Method java/lang/RuntimeException.getCause:()Ljava/lang/Throwable;
      34: checkcast     #9                  // class java/io/IOException
      37: athrow
      38: aload_2
      39: invokevirtual #8                  // Method java/lang/RuntimeException.getCause:()Ljava/lang/Throwable;
      42: instanceof    #10                 // class javax/security/auth/callback/UnsupportedCallbackException
      45: ifeq          56
      48: aload_2
      49: invokevirtual #8                  // Method java/lang/RuntimeException.getCause:()Ljava/lang/Throwable;
      52: checkcast     #10                 // class javax/security/auth/callback/UnsupportedCallbackException
      55: athrow
      56: aload_2
      57: athrow
      58: return
    Exception table:
       from    to  target type
           0    16    19   Class java/lang/RuntimeException

  static javax.security.auth.callback.CallbackHandler access$100(org.jboss.as.protocol.ProtocolConnectionUtils$WrapperCallbackHandler);
    Code:
       0: aload_0
       1: getfield      #1                  // Field wrapped:Ljavax/security/auth/callback/CallbackHandler;
       4: areturn
}
